mmv1/provider/terraform.go

```diff
@@ -507,10 +510,74 @@ func (t Terraform) CompileFileList(outputFolder string, files map[string]string,
 			continue
 		}
 		t.replaceImportPath(outputFolder, target)
+		if filepath.Ext(targetFile) == ".go" || filepath.Ext(targetFile) == ".markdown" {
+			t.addCopyfileHeader(source, outputFolder, target)
+		}
 		t.addHashicorpCopyRightHeader(outputFolder, target)
 	}
 }
 
+func (t Terraform) addCopyfileHeader(srcpath, outputFolder, target string) {
+	githubPrefix := "https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/"
+	if !strings.HasPrefix(srcpath, githubPrefix) {
+		srcpath = githubPrefix + srcpath
+	}
+
+	targetFile := filepath.Join(outputFolder, target)
+	sourceByte, err := os.ReadFile(targetFile)
+	if err != nil {
+		log.Fatalf("Cannot read file %s to add copy file header: %s", targetFile, err)
+	}
+
+	srcStr := string(sourceByte)
+	if strings.Contains(srcStr, "***     AUTO GENERATED CODE    ***    Type: Handwritten     ***") {
+		return
+	}
+
+	templateFile := "templates/terraform/handwritten_file.go.tmpl"
```

Comment:
Okay, may bad for missing the syntax difference between Go and Markdown files. Now, thinking about it more, managing two templates could create extra maintenance work. Your initial idea of using inline code seems a better solution, and it's also consistent with how the generated files are handled.  https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/api/resource.go#L1895
I think we should go back to that. Apologies for it!

Reply:
Done